File: Ahorn/entities/ELSTerminaHealth.jl
module ELSTerminaHealth

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/ELSTerminaHealth" ELSTerminaHealth(x::Integer, y::Integer, hardMode::Bool=false, maxHealth::Integer=300)

const placements = Ahorn.PlacementDict(
    "default" => Ahorn.EntityPlacement(ELSTerminaHealth)
)

function Ahorn.selection(entity::ELSTerminaHealth)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::ELSTerminaHealth, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/ELS_Termina_Health.lua
local utils = require("utils")

local elsTerminaHealth = {}

elsTerminaHealth.name = "MaggyHelper/ELSTerminaHealth"
elsTerminaHealth.depth = -100000
elsTerminaHealth.placements = {
    name = "default",
    data = {
        maxHealth = 300,
        hardMode = false
    }
}

elsTerminaHealth.fieldInformation = {
    maxHealth = {
        fieldType = "number",
        editable = true
    },
    hardMode = {
        fieldType = "boolean"
    }
}

elsTerminaHealth.fieldOrder = {
    "x",
    "y",
    "maxHealth",
    "hardMode"
}

function elsTerminaHealth.selection(room, entity)
    return utils.rectangle(entity.x - 50, entity.y - 20, 100, 40)
end

return elsTerminaHealth
